types/theme.d.ts

```diff
@@ -83,9 +83,11 @@ export interface ComponentStyleFunctionParam {
   variables: ComponentVariablesObject
 }
 
-export type ComponentPartStyleFunction = (styleParam?: ComponentStyleFunctionParam) => ICSSInJSStyle
+export type ComponentPartStyleFunction =
+  | ((styleParam?: ComponentStyleFunctionParam) => ICSSInJSStyle)
+  | undefined
 
-export type ComponentPartStyle = ComponentPartStyleFunction | ICSSInJSStyle
+export type ComponentPartStyle = ComponentPartStyleFunction | ICSSInJSStyle | undefined
```

Comment:
I'm still not sure this is required.  This doesn't exist on master yet typings compile, tests pass, and all functionality is working.

Could you recap why we need this again?

Reply:
Whenever you have compiler option `"strictNullChecks": true`, then it will complain.
This happens only because there is an indexed property defined together with optional. If you use one of them it is fine even with the compiler option enabled

![image](https://user-images.githubusercontent.com/5530260/44149199-717ba1d2-a09b-11e8-8cd5-0c68d7b1b2fa.png)